src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1347,6 +1347,45 @@ public IEnumerable<(string Prefix, long Value)> ReadIdentities<T>(TransactionOpe
             }
         }
 
+        public long GetNumberOfIdentities(TransactionOperationContext context, string databaseName)
+        {
+            var identities = context.Transaction.InnerTransaction.ReadTree(Identities);
+            var prefix = UpdateValueForDatabaseCommand.GetStorageKey(databaseName, null);
+
+            return GetNumberOf(identities, prefix, context);
+        }
+
+        public long GetNumberOfCompareExchange(TransactionOperationContext context, string databaseName)
+        {
+            var items = context.Transaction.InnerTransaction.OpenTable(CompareExchangeSchema, CompareExchange);
+            var compareExchange = items.GetTree(CompareExchangeSchema.Key);
+            var prefix = CompareExchangeCommandBase.GetActualKey(databaseName, null);
+            return GetNumberOf(compareExchange, prefix, context);
+        }
+
+        private static long GetNumberOf(Tree tree, string prefix, TransactionOperationContext context)
```

Comment:
Probably better to implement this explicitly in the Tree as `GetNumberOfEntriesWithPrefix` 
Like `GetNumberOfEntriesAfter` in Fixed Size Tree.